### PR TITLE
Adding model evaluation and plotting

### PR DESCRIPTION
**Pull request by elenafillo**
Improved model training and evaluation, including plotting. See `predicting.ipynb` for examples!
Updates:
- general training function `train_baseline_model()`
- general prediction function to generate baseline labels from a trained model `predict_baselines` 
- object `BaselineLabelledObservations` that aligns molefractions for a particular site and species with baseline labels, and does 
    - evaluating (e.g. labelled_obs.get_prediction_scores)
    - plotting, like `labelled_obs.plot_obs` and `labelled_obs.plot_obs_with_labels`:
image
    -calculates the monthly means timeseries `calculate_monthly_means` and quantifies anomalies, as well as plotting
image

- added option in config file to declare a `full_period` of available data for a site, different from the testing set
- added `predicting.ipynb` notebook with examples
- added test for predict_baselines

### CODE REVIEW

File: data/site_info.json [hunk 1]
```diff
@@ -4,7 +4,8 @@
         "coords": [53.3267, -9.9046],
         "training_period": [2016, 2018],
         "validation_period": [2019, 2019],
-        "testing_period": [2020, 2023]
+        "testing_period": [2020, 2023],
+        "full_period": [2010, 2023]
```
Comment (mrghg) [inline data/site_info.json:8]: Why is the full_period needed? In any case, it's only entered for MHD at the moment

File: ml_baselines/modelling/plot.py [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+from sklearn.metrics import confusion_matrix, ConfusionMatrixDisplay
+
+from matplotlib.colors import LinearSegmentedColormap
+
+cfg = Config()
+
+models_path = cfg.models_path
+
+def plot_confusion_matrix(y, y_pred, normalise=False, title="", labels = ["non-baseline", "baseline"]):
+
```
Comment (mrghg) [inline ml_baselines/modelling/plot.py:30]: Docstring needed

File: ml_baselines/modelling/plot.py [hunk 3]
```diff
@@ -0,0 +1,285 @@
...
+    disp.ax_.set_title(title)
+    disp.ax_.set(xlabel='Predicted label', ylabel='InTEM label')
+
+
+def plot_obs_with_labels(labelled_df, shade_train_and_val_periods=True, title="", site=None, plot_true_negatives=True):
+    """
+    Plot observed molefractions and predicted baseline labels over time, using confusion matrix labels (true positive, false positive etc)
+
+    Parameters:
```
Comment (mrghg) [inline ml_baselines/modelling/plot.py]: Not a big deal, but these docstrings are a different format to the ones in e.g., train.py

File: data/site_info.json (hunk 1 above)
Comment (mrghg) [reply data/site_info.json:8]: Wouldn't it be better to avoid hard-wiring this?

**Review by mrghg — APPROVED** [review]
Some minor changes suggested. Also, note co-pilot's suggestions, which seem sensible (especially adding a couple of tests). 

I like the idea of the class to store the predictions. I wonder if it'd be cleaner to restructure everything around one or two classes (in a future PR...)

File: data/site_info.json (hunk 1 above)
Comment (elenafillo) [reply data/site_info.json:8]: In the current setup, the evaluation statistics are calculated on the test set, but the code is not able to do testing on different time periods  (e.g. `[2010,2016], [2020,2023]` ), or for a testing period that includes the train/validation. Full period is used for plotting only, to recreate the paper's figures of all available data. A future update should enable one of the two complex testing modes above!
